ai-catalog/samples/magic-selfie/src/main/java/com/android/ai/samples/magicselfie/ui/MagicSelfieScreen.kt

```diff
@@ -19,202 +19,243 @@ import android.annotation.SuppressLint
-                },
-                enabled = (uiState !is MagicSelfieUiState.RemovingBackground) &&
-                    (uiState !is MagicSelfieUiState.GeneratingBackground),
-            ) {
-                Icon(Icons.Default.SmartToy, contentDescription = "Robot")
-                Text(modifier = Modifier.padding(start = 8.dp), text = "Generate")
-            }
 
-            if (uiState is MagicSelfieUiState.RemovingBackground) {
-                Spacer(
-                    modifier = Modifier
-                        .height(30.dp)
-                        .padding(12.dp),
-                )
-                Text(
-                    text = stringResource(R.string.removing_background),
-                )
-            } else if (uiState is MagicSelfieUiState.GeneratingBackground) {
-                Spacer(
+            if (selfieBitmap == null) {
+                PrimaryButton(
+                    text = stringResource(R.string.add_image),
+                    icon = painterResource(id = com.android.ai.uicomponent.R.drawable.ic_ai_img),
                     modifier = Modifier
-                        .height(30.dp)
-                        .padding(12.dp),
+                        .height(96.dp)
+                        .padding(start = 24.dp, end = 24.dp)
+                        .align(Alignment.Center),
+                    onClick = onTakePictureClick,
                 )
-                Text(
-                    text = stringResource(R.string.generating_new_background),
+            } else {
+                val displayBitmap = if (uiState is MagicSelfieUiState.Success) {
+                    uiState.bitmap
+                } else {
+                    selfieBitmap
+                }
+
+                Image(
+                    bitmap = displayBitmap.asImageBitmap(),
+                    contentDescription = null,
+                    contentScale = ContentScale.Crop,
+                    modifier = Modifier.fillMaxSize(),
                 )
-            } else if (uiState is MagicSelfieUiState.Error) {
-                val errorState = uiState as MagicSelfieUiState.Error
-                Spacer(
+
+                val textFieldState = rememberTextFieldState()
+                val keyboardController = LocalSoftwareKeyboardController.current
+
+                TextInput(
+                    state = textFieldState,
+                    placeholder = stringResource(R.string.prompt_placeholder),
                     modifier = Modifier
-                        .height(30.dp)
-                        .padding(12.dp),
-                )
-                Text(
-                    text = errorState.message ?: stringResource(R.string.unknown_error),
-                    color = MaterialTheme.colorScheme.error,
+                        .padding(10.dp)
+                        .height(80.dp)
+                        .align(Alignment.BottomCenter),
+                    primaryButton = {
+                        GenerateButton(
```

Comment:
doesn't have to be addressed in this PR, but perhaps we should have GenerateButton and SecondaryButton's `text` parameter be optional, similar to how `icon` is optional right now